MapboxNavigation/UserCourseView.swift

```diff
@@ -113,5 +113,143 @@ class UserPuckCourseView: UIView, UserCourseView {
             puckArrow?.setAffineTransform(CGAffineTransform.identity.rotated(by: -CGFloat(angle.toRadians())))
         }
     }
+}
+
+class NewUserPuckCourseView: UIView, UserCourseView {
+    
+    var puckLayerView: UserPuckStyleKitView!
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+    
+    func commonInit() {
+        puckLayerView = UserPuckStyleKitView(frame: bounds)
+        backgroundColor = .clear
+        puckLayerView.backgroundColor = .clear
+        addSubview(puckLayerView)
+    }
+    
+    var shouldAnimateTransformations: Bool = false
+    
+    var location: CLLocation?
     
+    var pitch: CLLocationDegrees = 0 {
+        didSet {
+            if oldValue != pitch {
+                let newPitch = pitch
+                UIView.animate(withDuration: 1, delay: 0, options: [.beginFromCurrentState, .curveLinear], animations: {
+                    var transform = CATransform3DRotate(CATransform3DIdentity, CGFloat(CLLocationDegrees(newPitch).toRadians()), 1.0, 0, 0)
+                    transform.m34 = -1.0 / 1000 // (-1 / distance to projection plane)
```

Comment:
@1ec5 would be cool to do something similar in `MGLAnnotationView` but it should be based on the actual distance from the camera to each individual annotation view.

Reply:
Yes, that’s something I explored in mapbox/mapbox-gl-native#5245. Though it seems like mbgl should expose the actual transform matrix it uses (to transform features in circle layers, for example), to reduce the guesswork.